test/factories/oauth_applications.rb

```diff
@@ -18,19 +18,19 @@
         # this line takes care of tests creating signin in order to look complete or modify delegatable on it.
         app.signin_permission.update(delegatable: false) && next if permission_name == SupportedPermission::SIGNIN_NAME
 
-        create(:supported_permission, application_id: app.id, name: permission_name)
+        create(:supported_permission, application: app, name: permission_name)
       end
 
       evaluator.with_supported_permissions_not_grantable_from_ui.each do |permission_name|
         next if permission_name == SupportedPermission::SIGNIN_NAME
 
-        create(:supported_permission, application_id: app.id, name: permission_name, grantable_from_ui: false)
+        create(:supported_permission, application: app, name: permission_name, grantable_from_ui: false)
```

Comment:
I appreciate you didn't introduce this change, but if we were to name the block parameter`application` we could pass `application:` instead of `application: app` here (and elsewhere in this commit) to further simplify things.

Reply:
It's a good thought, but I'm a bit nervous that it might be confusing given the factory is called `application` too. So I think I'll leave it for now. Also I'd prefer to get rid of the `after(:create)` hook altogether and I've got a personal to-do note about that!